### ISSUE TITLE
500 null pointer error from /blocks

### ISSUE THREAD

**Issue by Slord6** [nilsgawlik/gdmc_http_interface#1]
Mod Version: 0.4.0
Minecraft Version: 1.16.5

 I'm managing to cause a null pointer exception when PUTting blocks to the `/blocks` endpoint with the following data:

```Text

~0 ~0 ~0 minecraft:gold_block
~1 ~0 ~0 minecraft:iron_block
~2 ~0 ~0 minecraft:diamond_block
~3 ~0 ~0 minecraft:oak_stairs[facing=east, half=top, waterlogged=true,]
```

The client is written in C#, so not sure how useful the offending code to you is but I've written a min-repro:
```C#
/// <summary>
/// Push the given blocks to the server to be set
/// </summary>
public static void SetBlocksSync()
{
    using (WebClient webClient = new WebClient())
    {
        string body = "\n~0 ~0 ~0 minecraft:gold_block\n~1 ~0 ~0 minecraft:iron_block\n~2 ~0 ~0 minecraft:diamond_block\n~3 ~0 ~0 minecraft:oak_stairs[facing=east, half=top, waterlogged=true,]";
        string address = "http://localhost:9000/blocks?x=4&y=76&z=1";
        Debug.Log(address);
        Debug.Log(body);

        string result = webClient.UploadString(address, WebRequestMethods.Http.Put, body);
    }
}
```

I have set up a nodejs proxy to get some more visibility of what happens in the hopes that helps but nothing seems out of the ordinary.  The log should be fairly self-descriptive, but I can provide the proxy for reference if that's useful:

```
Proxying for client to: /blocks?x=4&y=76&z=1
With body '
~0 ~0 ~0 minecraft:gold_block
~1 ~0 ~0 minecraft:iron_block
~2 ~0 ~0 minecraft:diamond_block
~3 ~0 ~0 minecraft:oak_stairs[facing=east, half=top, waterlogged=true,] '
and headers  {
  'content-length': '165',
  expect: '100-continue',
  host: 'localhost:3000'
}
 and method PUT
Got proxied request response
.
Response: Internal server error: java.lang.NullPointerException
Status: 500 Internal Server Error
```

I haven't been able to reproduce in other clients but using the chrome console have seen behaviour where the server never returns a full response, if you think this is related I can provide that code

**Comment by Slord6** [comment]
Just realised this seems to happen with any blocks and with the simpler syntax for a single block.

```Text
Proxying for client to: /blocks?x=4&y=76&z=1
With body ' minecraft:stone '
and headers  {
  'content-length': '15',
  expect: '100-continue',
  host: 'localhost:3000'
}
and method PUT
Got proxied request response
.
Response: Internal server error: java.lang.NullPointerException
Status: 500 Internal Server Error
```

**Comment by nikigawlik** [comment]
@Slord6 Thanks for the detailed bug report! 

I was able to reproduce the issue (I use Unity myself, so the code was very useful).
The problem is that the interface naively expects an 'Accept' header to exist (for the new json functionality), so when it doesn't exist... NullPointerException. Kind of an oversight on my side, but in my defense many clients/http libaries set the Accept header automatically.

So for now you can just add the line `webClient.Headers.Set("Accept", "*/*");` in your C# code to fix the issue, until I get around to make a new release which fixes the issue on the backend.

**Comment by Slord6** [comment]
Ah, great - thanks!

**Comment by nikigawlik** [comment]
Fixed in [0.4.1 release](https://github.com/nilsgawlik/gdmc_http_interface/releases/tag/v0.4.1)
